gematria/datasets/process_and_filter_bbs.cc

```diff
@@ -0,0 +1,166 @@
+// Copyright 2024 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <fstream>
+#include <limits>
+#include <mutex>
+
+#include "X86.h"
+#include "X86InstrInfo.h"
+#include "X86RegisterInfo.h"
+#include "gematria/llvm/disassembler.h"
+#include "gematria/llvm/llvm_architecture_support.h"
+#include "gematria/utils/string.h"
+#include "llvm/ADT/StringExtras.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/Error.h"
+#include "llvm/Support/ThreadPool.h"
+
+using namespace llvm;
+
+cl::OptionCategory ProcessFilterCat("process_and_filter_bbs options",
+                                    "The options specifically for controlling "
+                                    "the behavior of process_and_filter_bbs.");
+
+static cl::opt<std::string> InputFile(
+    "input-file",
+    cl::desc("Path to the input CSV file containing hex basic blocks"),
+    cl::init(""), cl::cat(ProcessFilterCat));
+
+static cl::opt<std::string> OutputFile(
+    "output-file",
+    cl::desc(
+        "Path to the output CSV file with processed/filtered basic blocks"),
+    cl::init(""), cl::cat(ProcessFilterCat));
+
+static cl::opt<bool> FilterMemoryAccessingBlocks(
+    "filter-memory-accessing-blocks",
```

Comment:
I'd change this to `remove-memory-accessing-blocks` to make it immediately obvious that memory-accessing blocks are removed, not the only ones preserved.